projects/dot-custom-elements/src/components/dot-checkbox/dot-checkbox.tsx

```diff
@@ -0,0 +1,58 @@
+import { Component, Prop, State, Event, EventEmitter } from '@stencil/core';
+import { generateId } from '../../utils/utils';
+import Fragment from 'stencil-fragment';
+
+@Component({
+    tag: 'dot-checkbox',
+    styleUrl: 'dot-checkbox.scss'
+})
+export class DotCheckboxComponent {
+    @Prop() label: string;
+    @Prop() hint: string;
+    @Prop() options: string;
+    @Prop() value: string;
+    @Event() onCallback: EventEmitter;
+
+    @State() _options: any;
+    @State() _value: string;
+    _label: string;
+    _values = {};
+
+    componentWillLoad() {
+        this._options = this.options
```

Comment:
The format of the value for select, checkbox, radio button, etc is: `Label|value`